index.src.html

```diff
@@ -3772,6 +3780,26 @@
 {{ImageBitmapOptions}} {{ImageBitmapOptions/colorSpaceConversion}}. Setting this
 value to "none" disables color space conversion.
 
+### WebGPU ### {#videoframe-webgpu}
+
+WebCodecs extends [[WebGPU]] to allow importing {{VideoFrame}} objects via
+{{GPUDevice/importExternalTexture()}}, producing opaque {{GPUExternalTexture}}
+objects, which can be read from WebGPU shaders.
+The definition of the {{GPUExternalTextureDescriptor/source}} member of
+{{GPUExternalTextureDescriptor}} is modified to allow {{VideoFrame}}:
+
+<xmp>
+partial dictionary GPUExternalTextureDescriptor {
+    required (HTMLVideoElement or VideoFrame) source;
+};
+</xmp>
+
+The lifetime of such a {{GPUExternalTexture}} object is implicitly tied to the
+lifetime of the source {{VideoFrame}} object.
+When the {{VideoFrame}} is [=Close VideoFrame|closed=] (implicitly or
+explicitly), the {{GPUExternalTexture}} is *destroyed*, so no further operations
```

Comment:
> implicitly or explicitly

This should probably be in terms of the effect of close(), which is the clearing of the resource reference. We don't mention implicit close() elsewhere in the spec.

Any suggestions for how best to do that from the spec editors?

Reply:
"Close VideoFrame" is also invoked by transfers. I can explicitly call that out, or just not go into details.

Strictly speaking, there probably should be some step inside "Close VideoFrame" that does this. But I wasn't sure if it would be appropriate to invade on other parts of the spec for this.

Reply:
Text you settled on (mentioning transfer and close()) works for me. I'm also happy to support adding steps to "Close Video Frame" if you'd prefer. IIUC, this would be something like: for each texture referencing this frame, set textures [[detached]] to true.

Reply:
Text you settled on (mentioning transfer and close()) works for me. I'm also happy to support adding steps to "Close Video Frame" if you'd prefer. IIUC, this would be something like: for each texture referencing this frame, set textures [[detached]] to true.

Reply:
I'm OK with keeping this localized to this section if you are.

Reply:
Works for me.